frontend/src/app/products/show-details/show-details.component.html

```diff
@@ -61,11 +61,7 @@ <h2 >Delivery</h2>
                 No delivery possible
             </p>
         </div>
-        <button *ngIf="userService.loggedIn" color="primary" mat-flat-button (click)="goToBuy(product)">
-            Buy
-        </button>
-        <button color="primary" mat-flat-button (click)="goBack()">
-            Back
-        </button>
+        <app-action-buttons *ngIf="pageType === 'show'"></app-action-buttons>
```

Comment:
is ok. I think we can improve this by restructuring the component hierarchy a bit.. let's discuss this in a call